tests/components/mqtt/test_common.py

```diff
@@ -936,6 +936,9 @@ async def _test_encoding(
         hass, f"homeassistant/{domain}/item3/config", json.dumps(config3)
     )
     await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
```

Comment:
This is fine for now, but ideally we use a future/event to synchronize

Reply:
Not sure that was needed though, I probably forgot to remove the lines after testing.

Reply:
Addressed with: https://github.com/home-assistant/core/pull/113002